### PR TITLE
[FLINK-31807] Include all test code in architecture tests 

### PR DESCRIPTION
**Pull request by zentol**

### CODE REVIEW

File: flink-connector-cassandra/archunit-violations/dc1ba6f4-3d84-498c-a085-e02ba5936201 [hunk 1]
```diff
@@ -1,6 +1,12 @@
-org.apache.flink.streaming.connectors.cassandra.CassandraConnectorITCase does not satisfy: only one of the following predicates match:\
+org.apache.flink.connector.cassandra.source.CassandraSourceITCase does not satisfy: only one of the following predicates match:\
```
Comment (zentol) [inline flink-connector-cassandra/archunit-violations/dc1ba6f4-3d84-498c-a085-e02ba5936201:1]: FLINK-31804 will resolve this violation.

**Comment by echauchot** [comment]
@zentol taking a look after lunch break

File: flink-connector-cassandra/archunit-violations/dc1ba6f4-3d84-498c-a085-e02ba5936201 (hunk 1 above)
Comment (echauchot) [reply flink-connector-cassandra/archunit-violations/dc1ba6f4-3d84-498c-a085-e02ba5936201:1]: > [FLINK-31804](https://issues.apache.org/jira/browse/FLINK-31804) will resolve this "violation".

Once the other PR is merged and the cassandra connector uses the new rule, only `CassandraSourceITCase` "violation" will be removed. `CassandraConnectorITCase` "violation" will remain as this class does not use MiniCluster.

**Review by echauchot — APPROVED** [review]
thanks. LGTM

**Pull request merged**